Content.Server/GameObjects/Components/Kitchen/KitchenSpikeComponent.cs

```diff
@@ -27,50 +27,71 @@ void IActivate.Activate(ActivateEventArgs eventArgs)
         {
             SpriteComponent? sprite;
 
-            if (!EntitySystem.Get<SharedPullingSystem>().TryGetPulled(eventArgs.User, out var victim))
+            if (_meatParts == 0)
             {
-                if (_meatParts == 0)
-                {
-                    return;
-                }
-                _meatParts--;
+                return;
+            }
+            _meatParts--;
 
-                if (!string.IsNullOrEmpty(_meatPrototype))
+            if (!string.IsNullOrEmpty(_meatPrototype))
+            {
+                var meat = Owner.EntityManager.SpawnEntity(_meatPrototype, Owner.Transform.Coordinates);
+                if (meat != null)
                 {
-                    var meat = Owner.EntityManager.SpawnEntity(_meatPrototype, Owner.Transform.Coordinates);
-                    if (meat != null)
-                    {
-                        meat.Name = _meatName;
-                    }
+                    meat.Name = _meatName;
                 }
+            }
 
-                if (_meatParts != 0)
-                {
-                    eventArgs.User.PopupMessage(_meatSource1p);
-                }
-                else
+            if (_meatParts != 0)
+            {
+                eventArgs.User.PopupMessage(_meatSource1p);
+            }
+            else
+            {
+                if (Owner.TryGetComponent(out sprite))
                 {
-                    if (Owner.TryGetComponent(out sprite))
-                    {
-                        sprite.LayerSetState(0, "spike");
-                    }
-
-                    eventArgs.User.PopupMessage(_meatSource0);
+                    sprite.LayerSetState(0, "spike");
                 }
 
-                return;
+                eventArgs.User.PopupMessage(_meatSource0);
             }
 
+            return;
+            
+        }
+
+        public override bool DragDropOn(DragDropEventArgs eventArgs)
+        {
+            _ = TrySpike(eventArgs.Dragged, eventArgs.User);
+            return true;
+        }
+
+        public async Task<bool> TrySpike(IEntity victim, IEntity user)
```

Comment:
You should check for butcherable and meatparts up front as well as otherwise you can continuously drag stuff onto it that can't be butchered. Maybe also make it longer given it's a 3 second kill.